src/problem_bank_scripts/problem_bank_scripts.py

```diff
@@ -1230,3 +1230,10 @@ def backticks_to_code_tags(html):
     html = html.replace("\\`", "`")  # Replace escaped backticks
     return html
 
+def validate_header(header_dict):
+    url = "https://raw.githubusercontent.com/open-resources/learning_outcomes/main/Masterlist.csv"
+    learning_outcomes = pd.read_csv(url)
+    # check if header is sanctioned
+    topic_list = learning_outcomes["Topic"].unique().tolist()
```

Comment:
```suggestion
    topic_list = set()
    for subject in ["physics",] #"datasience", "statistics"]
        url = f"https://raw.githubusercontent.com/open-resources/learning_outcomes/main/outputs_csv/{subject}.csv"
        learning_outcomes = pd.read_csv(url)
        topic_list |= set(learning_outcomes["Topic"])
    # check if header is sanctioned
```
Using a set has a few advantages here, it keeps items in it unique, and then the check later on to see if a topic is in the set will be an O(1) operation, instead of requiring iteration over a list, meaning as more topics are added, this won't slow down as much, and also has a much cleaner syntax for expanding, with set operations ([docs here](https://docs.python.org/3.11/library/stdtypes.html?highlight=set#frozenset.update))

This also makes this script work with the changes Firas is preparing in open-resources/learning_outcomes#46, so while this will temporarily break this even more, it should mean we won't have to update it again once that PR lands, until the other topics get added (I don't know of a way to glob a directory on another repository, maybe someone else knows a way thats not a hacky workaround)

Though, maybe for now, we can do something like the following to make it easier to just adapt to adding more topic masterlists in the future:

```suggestion
    url = "https://raw.githubusercontent.com/open-resources/learning_outcomes/main/Masterlist.csv"
    learning_outcomes = pd.read_csv(url)
    topic_list = set(learning_outcomes["Topic"])
    # check if header is sanctioned
```

Reply:
Using a set makes sense, I'm going to stick with the second suggestion for now.
Is it better to read the csv files and save them to a variable outside the function? My thinking is that if there are enough of these checks being performed it is going to slow down the process by a considerable amount.

Reply:
I agree, thats a good idea to read and save them to variables them outside the function, though the only thing I'd be concerned about for putting the reads at the top level of the file (in no scope) would be a repeat of the PBH issues we had when github went down temporarily, but maybe thats realistically a nonissue